pkg/querier/queryrange/querysharding.go

```diff
@@ -138,17 +144,26 @@ func (ast *astMapperware) Do(ctx context.Context, r queryrange.Request) (queryra
 	case logqlmodel.ValueTypeStreams:
 		return &LokiResponse{
 			Status:     loghttp.QueryStatusSuccess,
-			Direction:  req.Direction,
-			Limit:      req.Limit,
-			Version:    uint32(loghttp.GetVersion(req.Path)),
+			Direction:  params.Direction(),
+			Limit:      params.Limit(),
+			Version:    uint32(loghttp.GetVersion(path)),
 			Statistics: res.Statistics,
 			Data: LokiData{
 				ResultType: loghttp.ResultTypeStream,
 				Result:     value.(loghttp.Streams).ToProto(),
 			},
 		}, nil
+	case parser.ValueTypeVector:
```

Comment:
I'm currently stuck here. `sharding.DownstreamEvaluator.Donwstream` returns a Matrix which ends up as a Vector with null points here :shrug:

Reply:
Thanks for digging into this. I'm not seeing empty responses anymore. The issue was that the `MatrixStepper` compares timestamps (https://github.com/grafana/loki/blob/main/pkg/logql/matrix.go#L46) and fills i empty points. Not really sure why. However, once I set the proper time stamp in my test setup it works.